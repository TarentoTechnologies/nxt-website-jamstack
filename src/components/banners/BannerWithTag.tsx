import transparentPNG from "../../../static/images/pngimg.com - laptop_PNG101775.png";
import {
  bgBlack,
  bgPallidBlueViolet2,
  white87,
} from "../../styles/style-guide/ColorGuide.module.css";
import { tagLabel2 } from "../../styles/style-guide/Typography.module.css";
import { PrimaryTitle } from "../titles/PrimaryTitle";
import {
  bannerImg,
  bannerWithTag,
  tag,
  tagPos,
} from "./BannerWithTag.module.css";

interface BannerWithTagProps {
  title: string;
  tagsList: any;
}

export const BannerWithTag = ({ title, tagsList }: BannerWithTagProps) => {
  return (
    <div
      className={`container-fluid ${bgPallidBlueViolet2} ${bannerWithTag} h-100 position-relative`}
    >
      <div className="row">
        <div className="col-sm-12 col-md-12 col-lg-12 col-xl-6 p-5">
          <PrimaryTitle title={title} variant="infoSectionH2" />
          {/* For larger screens */}
          <div
            className={`row gy-3 position-absolute ${tagPos} d-none d-sm-none d-md-flex`}
          >
            {tagsList?.map((data: any, inde: number) => {
              return (
                <div
                  className={`${tag} ${bgBlack} d-flex align-items-center justify-content-center col-sm-12 col-md-3 col-lg-3 col-xl-3 me-3`}
                  key={data.id}
                >
                  <label className={`${tagLabel2} ${white87}`}>
                    {data.Label}
                  </label>
                </div>
              );
            })}
          </div>

          {/* For smaller screens */}
          <div className={`row gy-3 ${tagPos} d-flex d-sm-flex d-md-none mt-5`}>
            {tagsList?.map((data: any, index: number) => {
              return (
                <div
                  className={`${tag} ${bgBlack} d-flex align-items-center justify-content-center  col-sm-3 col-md-3 col-lg-3 col-xl-3 me-3`}
                  key={data.id}
                >
                  <label className={`${tagLabel2} ${white87}`}>
                    {data.Label}
                  </label>
                </div>
              );
            })}
          </div>
        </div>
        <div className="col-sm-12 col-md-12 col-lg-12 col-xl-6 overflow-hidden p-0 m-0">
          <img
            src={transparentPNG}
            alt="Test Image"
            className={`img-fluid ${bannerImg} float-end mt-0 mt-sm-0 mt-md-4 mt-lg-4 mt-xl-4`}
          />
        </div>
      </div>
    </div>
  );
};
