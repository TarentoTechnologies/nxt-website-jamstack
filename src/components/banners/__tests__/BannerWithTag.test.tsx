import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { BannerWithTag } from "../BannerWithTag";

const data = {
  title: "The Design tools and Technology that we use to support CIBoost",
  tagsList: [
    {
      id: 0,
      Label: "Sketch",
    },
    {
      id: 1,
      Label: "Photoshop",
    },
    {
      id: 2,
      Label: "HTML",
    },
    {
      id: 3,
      Label: "React",
    },
    {
      id: 4,
      Label: "PHP",
    },
  ],
};

const imgData = {
  childImageSharp: {
    gatsbyImageData: {
      layout: "constrained",
      placeholder: {
        fallback:
          "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAQCAYAAAAWGF8bAAAACXBIWXMAAC4jAAAuIwF4pT92AAACvUlEQVR42q2UXUhTYRjHp+587wOkiwq8m05W4gcIMrFSp1teCF7sygK982ZBJNqFjQLtItM06WIXQSGOHS3namYq80wnJsSqOV2YSjey2UaU1TLynPP0nvm1wDGCHvid/3s4z/md933P4chkSWU2m7N0Oh1+gEajISRk+3mEiTCZEBYLwXGcXHZMZUgHgiT7MBz/iBPEEkaQK4jQHkSIQImuhwiKClE0HaIZ5QqjUK5lZ5/wNFosqmRPxoEVJ6lpkiSBoigeASlBPahPJAgC0ENiDQ2Np1IIySnpBpqmdxFiGgSpF4mjhvr606mE0/tCHgFpEOk9YcxobEg5w2mpmWEYHgFpEBmGRkIqer7OfDJZeFjohUxJzQqFgkdAGkQpSYqKGQx/L/lIiGEv/1mIlqzXG44XoiVPKpVKUKtUPALU6lSoJUQV6qFp6lhhYpApl7/ApU+BIHkpE+BJicAwHCUupSiXYyixaF3yHgJAQoaS6R8YWG5qbobKqhqh1lgHNbUmqDVehKoqA1RWGqDi3AXQ6yugrKwcSkpKxepqA1hvWXdn5ubK9x2Z0iFLOvka32n58v0HXLrcJOTkaKC4uBR0ukLIy9OBVnsWNJp8lGcgT6uDgoIiyM3VonE+z81y8GFtbdZms2GHexeJRJjt+M7y5qcYPBoc2u3tuy903u4SrDetwrXWVqHtervQ1t4mdNzoEDq7OoWe3m6h916P0H33juB0OX8HQyFY9PuvHgq34/EraKbw7ecvPhz7DKvrG/Am8A642RkYecoCO+KA4ScsOFg7OIbt4Ho+CmOuUZiYdIN3jhPfLgUgsByMezyewoQwvLXVEolGXZvhsH19Y4N9v7rKBoNB1u9/zfp8XsfCwrx9cfHVkNfrGXQ/G3087nY+dLudtonxsQecZ6p/Yd7b7fN5+tGfp0j2v+sPIReLz2Ra9xMAAAAASUVORK5CYII=",
      },
      images: {
        fallback: {
          src: "/static/c3c04e30a647b67d02651bf87dd55afe/f4954/pngimg_com_laptop_PNG_101775_c0ab79f5b3.png",
          srcSet:
            "/static/c3c04e30a647b67d02651bf87dd55afe/0fcdd/pngimg_com_laptop_PNG_101775_c0ab79f5b3.png 877w,\n/static/c3c04e30a647b67d02651bf87dd55afe/12422/pngimg_com_laptop_PNG_101775_c0ab79f5b3.png 1753w,\n/static/c3c04e30a647b67d02651bf87dd55afe/f4954/pngimg_com_laptop_PNG_101775_c0ab79f5b3.png 3506w",
          sizes: "(min-width: 3506px) 3506px, 100vw",
        },
        sources: [],
      },
      width: 3506,
      height: 2805,
    },
  },
};

describe("Test if the BannerWithTag component is working as expected", () => {
  it("<BannerWithTag /> matches snapshot", () => {
    const component = render(
      <BannerWithTag
        data-testid="banner-with-tag"
        title={data.title}
        tagsList={data.tagsList}
        img={imgData}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<BannerWithTag /> renders without exploding", () => {
    const component = render(
      <BannerWithTag
        data-testid="banner-with-tag"
        title={data.title}
        tagsList={data.tagsList}
        img={imgData}
      />
    );
    expect(component.getByTestId("banner-with-tag")).toBeInTheDocument();
  });
});
