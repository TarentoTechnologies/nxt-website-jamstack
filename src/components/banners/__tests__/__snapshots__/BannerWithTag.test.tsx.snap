// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the BannerWithTag component is working as expected <BannerWithTag /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid bgPallidBlueViolet2 bannerWithTag h-100 position-relative"
    data-testid="banner-with-tag"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-12 col-md-12 col-lg-12 col-xl-6 p-5"
      >
        <div
          class="
      undefined
        
        
        infoSectionH2
        
        
        
        
        
        
        
        
        "
          data-aos="fade-up"
          data-testid="primary-title"
        >
          The Design tools and Technology that we use to support CIBoost
        </div>
        <div
          class="row gy-3 position-absolute tagPos d-none d-sm-none d-md-flex"
        >
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center col-sm-12 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              Sketch
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center col-sm-12 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              Photoshop
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center col-sm-12 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              HTML
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center col-sm-12 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              React
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center col-sm-12 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              PHP
            </label>
          </div>
        </div>
        <div
          class="row gy-3 tagPos d-flex d-sm-flex d-md-none mt-5"
        >
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center  col-sm-3 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              Sketch
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center  col-sm-3 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              Photoshop
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center  col-sm-3 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              HTML
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center  col-sm-3 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              React
            </label>
          </div>
          <div
            class="tag bgBlack d-flex align-items-center justify-content-center  col-sm-3 col-md-3 col-lg-3 col-xl-3 me-3"
            data-aos="fade-up"
          >
            <label
              class="tagLabel2 white87"
              data-aos="fade-up"
            >
              PHP
            </label>
          </div>
        </div>
      </div>
      <div
        class="col-sm-12 col-md-12 col-lg-12 col-xl-6 overflow-hidden p-0 m-0"
      />
    </div>
  </div>
</div>
`;
